packages/core/src/language-server/binding.ts

```diff
@@ -119,20 +121,13 @@ export function bindLanguageServerPool({
         let language = server.getLanguageType(textDocument.uri);
         let formating = configManager.getFormatCodeSettingsFor(language);
         let preferences = configManager.getUserSettingsFor(language);
-        let diagnosticCodes = context.diagnostics;
-        // "only" is only present when a user hovers over and the codefix dropdown is requested. If it is missing the user has explicitly asked for codefixes
-        let type =
-          context.only === undefined
-            ? CodeActionKind.QuickFix
-            : context.only.includes(CodeActionKind.QuickFix)
-            ? CodeActionKind.QuickFix
-            : undefined;
+        let diagnostics = context.diagnostics;
 
         return server.getCodeActions(
           textDocument.uri,
-          type,
+          CodeActionKind.QuickFix,
```

Comment:
Generally curious about your thoughts here about passing the kind. For the vscode binding we only let QuickFixs requests but I'm trying to balance the fact that you can all `getCodeActions` programmatically.

Reply:
If I'm reading this right, this means that when the editor sends a request for _any_ kind of code actions, we'll always respond with quickfixes?

You've spent a lot more time thinking about this than I have, so I'm probably missing something, but my kneejerk reaction would be to say it's better to return an empty response for an action request we don't support than to give back something that the editor didn't ask for. Why doesn't [the short-circuit you have in the language server](https://github.com/chadhietala/glint/blob/chietala/ts-lang-server-plumbing/packages/core/src/language-server/glint-language-server.ts/#L386) already handle this the way we'd want?

Reply:
@dfreeman we solicit to vscode that we will only support [`QuickFix`](https://github.com/typed-ember/glint/pull/499/files#diff-60981cd0aadb151d3099bec522b460290b800c50f256090680ddc618b53d74daR30) so we should only get requests we can fulfill. The short circuit is there in the event you're using the server from `analyzeProject` directly.

Reply:
If a well-behaved editor will only ever ask us for quickfixes, why do we need to hard-code `QuickFix` here?

Reply:
I spent a day looking as to why [`context.only`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#codeActionContext) is optional and not always passed for CodeFix requests but found nothing. All implementations of language servers that I found ([Svelte](https://github.com/sveltejs/language-tools/blob/220abfbf10068d60d72410fe4aa8ecf99f6ad0af/packages/language-server/src/plugins/svelte/features/getCodeActions/index.ts#L20) [Vue](https://github.com/vuejs/vetur/blob/db9b6e70f4609e54a11ff456284ed01f79424f74/server/src/modes/script/javascript.ts#L976),  that provide codefixes do a narrowing like:

```ts
let type = '';

if (context.only === undefined) {
  type = CodeActionKind.QuickFix;
} else if (context.only.includes(CodeActionKind.QuickFix)) {
  type = CodeActionKind.QuickFix;
}
```

I can add this if you think it will be clearer.

Reply:
Thanks, that's super helpful context! I think something along those lines would help make it clearer for anyone (read: me in the future when I've forgotten this conversation 😅) who comes along and reads it later

Reply:
Sounds good. I pushed another commit that does ^ and explains whats going on with `context.only`